lib/components/base-components/NormalComponent.ts

```diff
@@ -115,6 +122,26 @@ export class NormalComponent<
           }
         }
       }
+      // pin label are string
+      if (normalizationNeeded) {
```

Comment:
not a big deal, but by convention i try to always have a prefix `is*` `are*` etc.

```suggestion
      if (isNormalizationNeeded) {
```